js/stack/stack-data-structure.js

```diff
@@ -5,22 +5,33 @@ class StackDataStructure {
   }
 
   canPush() {
-    // ... your code goes here
+      if (this.stackControl.length < this.MAX_SIZE) {
+        return true 
+      } else {
+        return false
+      }
   }
 
   display() {
     // ... your code goes here
   }
 
   isEmpty() {
-    // ... your code goes here
+  return this.stackControl.length === 0
   }
 
   push(item) {
-    // ... your code goes here
+    if (this.canPush() === true) {
+      this.stackControl.push(item)
+    } if (this.stackControl.length >= this.MAX_SIZE) {
+      return "Stack Overflow" 
+    }
```

Comment:
```suggestion
    if (this.canPush() === true) {
      this.stackControl.push(item)
      return this.stackControl
    } if (this.stackControl.length >= this.MAX_SIZE) {
      return "Stack Overflow" 
    }
```

Reply:
Únicamente te faltaba hacer el return de this.stackControl, para retornar el array con el nuevo elemento. Añadiendo eso ya tienes el jasmine de Stack totalmente en verde